tools/istio-iptables/pkg/capture/run_test.go

```diff
@@ -423,30 +429,22 @@ func setup(t *testing.T) {
 		assert.NoError(t, userns.WriteGroupMap(map[uint32]uint32{userns.OriginalGID(): 0}))
 		// Istio iptables expects to find a non-localhost IP in some interface
 		assert.NoError(t, exec.Command("ip", "addr", "add", "240.240.240.240/32", "dev", "lo").Run())
+		// Put a new file we have permission to access over xtables.lock
+		xtables := filepath.Join(t.TempDir(), "xtables.lock")
+		_, err := os.Create(xtables)
+		assert.NoError(t, err)
```

Comment:
Not sure why this logic was restored but I think it will introduce test errors in iptables-legacy environments due to the inability to write the lockfile inside /run when the test runs within the build container.
I would recomend to use the XTABLES_LOCKFILE override instead.

Right now, this error is hidden due to the filter probe, which fails because of failure in creating xtables.lock, letting the iptables-wrapper prefer iptables-nft instead, which does not use file-based locking.
If the tests is executed from a environment that only has iptables-legacy then the tests will likely fail.

Reply:
> Not sure why this logic was restored 

Without it, the tests were failing for me (in or out of container, I don't remember which) - I'll look more closely at why.

>  If the tests is executed from a environment that only has iptables-legacy then the tests will likely fail.

Yeah but we have no such environment in CI or within the `build-tools` container, and those are our primary targets + anyone can use the latter to run the  tests regardless of their local env.

Reply:
> Without it, these non-cni iptables tests were failing for me (in or out of container, I don't remember which) the tests wrapped in the namespace check do not - I'll look more closely at why.

This is weird since the CNI tests are also using the lockfile override. Maybe another problem is lurking.

Reply:
Ah, I think I was seeing a side effect of the detection bug before I fixed it, this reversion isn't required, so undoing this.

Reply:
Thank you for checking :)